src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimService.java

```diff
@@ -179,17 +177,17 @@ public Claim saveClaim(
 
         Claim issuedClaim;
         try {
-            Long prePaymentClaimId = caseRepository.getOnHoldIdByExternalId(externalId, authorisation);
-
-            LocalDateTime now = LocalDateTimeFactory.nowInLocalZone();
+            caseRepository.getClaimByExternalId(externalId, authorisation).ifPresent(claim -> {
```

Comment:
why do we need this? CCD will throw error if case is already created  (Please test locally ) as well as claim store throws conflict exception on save

Reply:
Currently, we are using this to make sure that there is no claim already created in claim store. This is replacement of previously existing check if any case is already on onhold status. Instead of check on hold, currently we are checking if there is any case there with same external reference id.

Reply:
In my opinion calling getByExternalId before first save to throw duplicate error, looks unnecessary. 
The underlying repository should throw duplicate error based on the unique index on external id. If there is no unique index on claim number or UUID we should request CCD team to add this one for our claims. 

Note: This additional call will have impact on performance which is already not good, so need to make a call what problem this call is going to solve at the cost of performance.

Reply:
@sabahirfan there is no implementation of CCD to enforce uniqueness on a field of claim. So only way is to do defensive check.

Reply:
Also the above contention is claimrepository, which will go away when we move to CCD as main.

Reply:
I'd added the `try...catch` block to ensure that the claim creation notifications were sent in a particular communication failure scenario. Whatever changes you make here, just ensure that the event is produced even if the claim already exists.

Reply:
good one @andrewwa-kainos . Will ensure. I am not changing the flow.

Reply:
@andrewwa-kainos We are handling the CONFLICT exception in `front-end`, I am not sure why we need to suppress this exception in backend. Please have a look at:
https://github.com/hmcts/cmc-citizen-frontend/blob/5810c52e916ed4ed3a8834633c88d169a42c5191/src/main/app/claims/claimStoreClient.ts#L77

Reply:
Because we still need to proceed with the claim issued event handling.

Reply:
That event must have already handled in the first successful save operation. As we rollback the transaction if any of those handlers fail.

Reply:
Either the Transaction scope is not properly done or we should check if the claim already exist before even taking it to Payment page from front end.